### ISSUE TITLE
Very high CPU load

### ISSUE THREAD

**Issue by janusn** [nicolargo/glances#2601]
**Describe the bug**
The CPU consumption of glances is very high. Disabling all plugins as suggested on https://github.com/nicolargo/glances/issues/1823#issuecomment-802179830 by @nicolargo makes no differences.

**To Reproduce**
1. Start Glances with the following options '-w --password -u ha --disable-plugin sensors,folders,connections'
2. monitor the resources with htop.
3. The CPU usage fluctuates between 20% to 100%.  See the screencap.

**Expected behavior**
A resource tool like glances should be relatively low resources demanding.

**Screenshots**
htop
<img width="1578" alt="Screen Shot 2023-11-04 at 6 46 40 pm" src="https://github.com/nicolargo/glances/assets/21253783/edac662f-4d5e-428c-be4b-3ff31681484f">

Glance webUI
<img width="1958" alt="Screen Shot 2023-11-04 at 7 15 45 pm" src="https://github.com/nicolargo/glances/assets/21253783/f8ab0109-4243-4c09-9913-ebb233e5feac">

**Environement (please complete the following information)**
 - Operating System
```
$ uname -a
    Linux DS918Plus 4.4.302+ #64570 SMP Thu Jul 20 00:07:29 CST 2023 x86_64 GNU/Linux synology_apollolake_918+
```
 - Hardware: Synology DS918+ Intel Celeron J3455 16GB RAM
 - Glances & psutil versions: `To be completed with result of: glances -V`
```
# glances -V
Glances v3.4.0.3 with PsUtil v5.9.5
Log file: /tmp/glances-root.log
```
 - How do you install Glances (Pypi package, script, package manager, source):
glances is installed from a docker container on image 'nicolargo/glances:latest-full'
 - docker version: 
```
$ docker -v
Docker version 20.10.23, build 876964a
``` 
 - Glances test (only available with Glances 3.1.7 or higher):
[glances --issue](https://pastebin.com/S6rxPsuT)

**Additional context**
I have another glances install on a docker container on a NUC with Intel Core i5 1340P 32GB RAM. The CPU usage on it is lower than the one running on DS918+, of course. Nonetheless, it still reports 20-30% of a core. That is CPU intensive to me. The consumption is about as much as a frigate handling 3 x 4K streams and object detection on another container on the same hardware.

You can also [pastebin](https://pastebin.com/):

* the Glances configuration file (https://glances.readthedocs.io/en/latest/config.html#location) [glances.conf](https://pastebin.com/h1AFGx87)
* the Glances log file (https://glances.readthedocs.io/en/latest/config.html#logging)
```
2023-11-04 19:31:59,127 -- INFO -- Start Glances 3.4.0.3
2023-11-04 19:31:59,128 -- INFO -- CPython 3.11.6 (/venv/bin/python3) and psutil 5.9.5 detected
2023-11-04 19:31:59,142 -- INFO -- Read configuration file '/etc/glances.conf'
2023-11-04 19:31:59,150 -- INFO -- Read password from file /root/.config/glances/ha.pwd
2023-11-04 19:31:59,201 -- INFO -- Start GlancesWebServer mode
2023-11-04 19:31:59,686 -- ERROR -- containers (Podman) plugin - Can't connect to Podman (http://%2Frun%2Fuser%2F1000%2Fpodman%2Fpodman.sock/v4.7.0/libpod/_ping (HEAD operation failed))
2023-11-04 19:31:59,915 -- ERROR -- Cannot grab battery status [Errno 2] No such file or directory: '/sys/class/power_supply'.
2023-11-04 19:31:59,926 -- ERROR -- Can not init battery class ([Errno 2] No such file or directory: '/sys/class/power_supply')
2023-11-04 19:31:59,960 -- WARNING -- Wifi lib is not compliant with Python 3, Wifi plugin is disabled
2023-11-04 19:32:00,791 -- INFO -- Glances RESTful API Server started on http://0.0.0.0:61208/api/3/
2023-11-04 19:32:00,796 -- INFO -- Glances Web User Interface started on http://0.0.0.0:61208/
```

**Comment by janusn** [comment]
@nicolargo Thanks for your reply.

No, I didn’t change the config file.  I once tried to reduce the refresh frequency but it didn’t help at all. I reverted it back to the original value.

I have an observation which may or may not be relevant. Sometimes glances reports very low CPU usage of the glances container in the container section, like less than 1% but it reports the real CPU usage (~100%) of glances in the process section as “python3” without arguments.

**Comment by janusn** [comment]
@nicolargo 

Is there anything else I can help to identify the issue?

**Comment by peterpepo** [comment]
+1 for this issue.
Raspberry pi3b, running in docker, webui open with default 10s refresh rate results in ~30%cpu usage.

**Comment by nicolargo** [comment]
I am currently working on Glances v4 and i just solve an issue with the -u (WebUI basic authentication). This option generate a high CPU consumption. Can you try to run your container without this option to confirm the issue ?

**Comment by janusn** [comment]
Thanks for following it up.

I have tried to start the container glances with the following compose file on version, 3.6.1.2.

From the samples I observed from htop. It does not make any signicant difference. On an Intel Core i5 1340P, the average of the most resources hungry process consume about 32.8% of a single core on average over a minute with or without "-u" option. I've tried taken out the "--password" option as well. It lowers a bit to 31.8% but well within statistical errors.

```
---
version: "3"
services:
  glances:
    image: nicolargo/glances:latest-full
    container_name: glances
    deploy:
      resources:
        limits:
          memory: 512M
    pid: host
    network_mode: host
    environment:
      - "GLANCES_OPT=-C /config/glances.conf -w --password"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config:/config:ro
    secrets:
      - source: glances_password
        target: /root/.config/glances/ha.pwd
    restart: unless-stopped
secrets:
  glances_password:
    file: ./secrets/ha.pwd
```

**Comment by nicolargo** [comment]
Also remove the --password.

**Comment by janusn** [comment]
I have tried without the "--password" as well and I have mentioned in the last comment. It does not make much of a difference.

**Comment by nicolargo** [comment]
You can also give a chance to the dev image version (based on Glances
version 4).

Just be aware that Glances version 4 is under development. So do not use in
production.

**Comment by janusn** [comment]
I have tried to run the dev version. However, it encountered an issue.
```
$ sudo docker compose -f /docker/hass/hass.yaml up -d
[+] Running 1/1
 ⠋ Container glances     Creating                                                                                                                          0.0s 
Error response from daemon: unknown log opt 'max-file' for journald log driver

```
I think it caused by I am using journald as the log driver.
```
$ cat /etc/docker/daemon.json.2023-12-16 
{
  "log-driver": "journald"
  }
}
```
Should I report it as a separated issue?

**Comment by janusn** [comment]
I managed to get the dev version running by switching the log driver back to json-file.
Now the CPU usage of glances is much lower now. it hovers around 10.4%.

**Comment by janusn** [comment]
Btw, I have just discovered that the glances container behave differently on the 2 machines I run it on.
On ubuntu 23.10, there are 28 instances of glances running. (/venv/bin/python3 -m glances -C /config/glances.conf -w --password -u ha)

However, on Synology DSM 7.2-64570 Update 3, there is just a single instance.

Is it normal?

**Comment by matthijslinku** [comment]
I've noticed the same issue, but only when I run glances with the `--password` flag. Without it, the python3 process runs at ~30% on an rPI2b. With the flag it runs at 99%.

**Comment by nicolargo** [comment]
Yes this bug is corrected on de develop branch

Le mar. 16 janv. 2024, 15:11, matthijslinku ***@***.***> a
écrit :

**Comment by ynazar1** [comment]
I'll chime in here with my findings too. 
Running glances as `/usr/bin/python3 /usr/local/bin/glances -w` uses about 1 to 2% of my CPU on the server... Running glances like this: `/usr/bin/python3 /usr/local/bin/glances -w -u glances` spikes CPU usage to consistent ~30%.... 

Yes it's mostly on one core but that makes a huge difference on smaller servers, ie: older RasPIs. So until bug fix is released (and if you don't care too much about webserver security...), run it without username and CPU usage should be reasonable.

**Comment by nicolargo** [comment]
Solved in Glances 4.0.0 or higher
